lib/SIL/Parser/ParseSIL.cpp

```diff
@@ -1212,33 +1260,7 @@ bool SILParser::parseSILType(SILType &Result,
     return true;
   
   // Resolve the generic environments for parsed generic function and box types.
-  class HandleSILGenericParamsWalker : public ASTWalker {
```

Comment:
No. This is not the right way to do this. Here is my suggestion.

* Commit 1 - Create a private header called ./lib/SIL/Parser/SILParser-Private.h. Move the declaration, not definition of SILParser there. All .cpp stuff stays in same place.
* Commit 2 - Move this into a separate file called SILParseType.cpp.